packages/datatrak-web/src/features/Survey/Components/SurveyQuestion.tsx

```diff
@@ -70,44 +78,100 @@ export const SurveyQuestion = ({
   type,
   name,
   updateFormDataOnChange,
+  validationCriteria = {},
   ...props
 }: SurveyQuestionFieldProps) => {
-  const { control } = useFormContext();
-  const { setSingleAnswer } = useSurveyForm();
+  const { control, formData, errors } = useFormContext();
+  const { setFormData } = useSurveyForm();
   const FieldComponent = QUESTION_TYPES[type];
 
   if (!FieldComponent) {
     return <QuestionPlaceholder>{name}</QuestionPlaceholder>;
   }
 
+  const controllerName = getNameForController(name, type);
+
   // If the question dictates the visibility of any other questions, we need to update the formData when the value changes, so the visibility of other questions can be updated in real time. This doesn't happen that often, so it shouldn't have too much of a performance impact, and we are only updating the formData for the question that is changing, not the entire formData object.
   const handleOnChange = e => {
     if (updateFormDataOnChange) {
-      setSingleAnswer(name, e.target.value);
+      setFormData({
```

Comment:
What made you decide to take this out of the context reducer?

Reply:
I realised we didn't actually need it, because we the `setFormData` can handle an object with one key, and then working on the condition question type I ended up needing to do more updates to the data as anything changed, and didn't want to call the same thing in multiple places and potentially end up with bugs. And since I was removing it that branch I just did it here too so I didn't get conflicts